omnigibson/robots/manipulation_robot.py

```diff
@@ -1055,21 +1049,13 @@ def _get_assisted_grasp_joint_type(self, ag_obj, ag_link):
             return None
         
         # Otherwise, compute the joint type. We use a fixed joint unless the link is a non-fixed link.
+        # A link is non-fixed if it has any non-fixed parent joints.
         joint_type = "FixedJoint"
-        if ag_obj.root_link != ag_link:
-            # We search up the tree path from the ag_link until we encounter the root (joint == 0) or a non fixed
-            # joint (e.g.: revolute or fixed)
-            link_handle = ag_link.handle
-            joint_handle = self._dc.get_rigid_body_parent_joint(link_handle)
-            while joint_handle != 0:
-                # If this joint type is not fixed, we've encountered a valid moving joint
-                # So we create a spherical joint rather than fixed joint
-                if self._dc.get_joint_type(joint_handle) != JointType.JOINT_FIXED:
-                    joint_type = "SphericalJoint"
-                    break
-                # Grab the parent link and its parent joint for the link
-                link_handle = self._dc.get_joint_parent_body(joint_handle)
-                joint_handle = self._dc.get_rigid_body_parent_joint(link_handle)
+        for edge in nx.edge_dfs(ag_obj.articulation_tree, ag_link.body_name, orientation="reverse"):
```

Comment:
I'm not too familiar with nx / graphs -- so when a joint gets created during AG, I'm assuming the articulation tree is automatically updated so that when it's queried here a valid joint edge should be found? Is this an efficient computation?

Reply:
This does not include the AG joint - it's the joints the object has at load time. It is an efficient computation because O(V+E) is on average ~1.2 or something I would guess.